### DISCUSSION TITLE
How use secp256k1 without libsecp256k1.dll

### DISCUSSION THREAD

**Discussion by MiZentUi** [bitcoin-core/secp256k1#1591] [Q&A] [answered]
I try to use static library but always use shared one. I tried to defind macro (#define SECP256K1_STATIC) in main file of my project but it didn't affect. I use cmake on windows.

**Comment by real-or-random** [comment, marked as answer]
> #define SECP256K1_STATIC) 

This sounds right. Do you do this before including `secp256k1.h`?

Do you actually build a static library? (Pass `-DBUILD_SHARED_LIBS=OFF` to cmake when building libsecp256k1?)  

What compiler are you using?

**Reply by MiZentUi** [reply]
Yes, before including and I link static library to project

**Reply by MiZentUi** [reply]
but I didn’t set the value in cmake, I’ll recompile
